File: Samples/Forms and Annotations/FlattenFormFields/C#/FlattenFormFields.cs
using System;
using System.Diagnostics;
using System.Linq;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class FlattenFormFields
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            const string pathToFile = "FlattenFormFields.pdf";

            bool flattenWidgets = false;
            bool flattenSomeOfTheControls = false;

            using (var pdf = new PdfDocument(@"..\Sample Data\form.pdf"))
            {
                pdf.FlattenControls();

                // Alternatively, you can:

                if (flattenWidgets)
                {
                    // 1. Flatten both controls and annotations
                    pdf.FlattenWidgets();
                }

                if (flattenSomeOfTheControls)
                {
                    // 2. Flatten individual controls
                    foreach (PdfControl control in pdf.GetControls().ToArray())
                        control.Flatten();
                }

                pdf.Save(pathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pathToFile) { UseShellExecute = true });
        }
    }
}

File: Samples/Forms and Annotations/FlattenFormFields/VB.NET/FlattenFormFields.vb
Imports BitMiracle.Docotic
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class FlattenFormFields
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Const pathToFile As String = "FlattenFormFields.pdf"

            Dim flattenWidgets = False
            Dim flattenSomeOfTheControls = False

            Using pdf As New PdfDocument("..\Sample Data\form.pdf")
                pdf.FlattenControls()

                ' Alternatively, you can

                If flattenWidgets Then
                    ' 1. Flatten both controls and annotations
                    pdf.FlattenWidgets()
                End If

                If flattenSomeOfTheControls Then
                    ' 2. Flatten individual controls
                    For Each control As PdfControl In pdf.GetControls().ToArray
                        control.Flatten()
                    Next
                End If

                pdf.Save(pathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(pathToFile) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace
